pkg/cephmgr/rgw/leader.go

```diff
@@ -0,0 +1,231 @@
+/*
+Copyright 2016 The Rook Authors. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package rgw
+
+import (
+	"fmt"
+	"path"
+
+	etcd "github.com/coreos/etcd/client"
+	ctx "golang.org/x/net/context"
+
+	"github.com/rook/rook/pkg/cephmgr/client"
+	"github.com/rook/rook/pkg/cephmgr/mon"
+	"github.com/rook/rook/pkg/clusterd"
+	"github.com/rook/rook/pkg/util"
+)
+
+const (
+	RGWKey         = "rgw"
+	ObjectStoreKey = "object"
+)
+
+type Leader struct {
+}
+
+func NewLeader() *Leader {
+	return &Leader{}
+}
+
+// Apply the desired state to the cluster. The context provides all the information needed to make changes to the service.
+// Initialize CephFS. Must be idempotent.
+func (r *Leader) Configure(context *clusterd.Context, factory client.ConnectionFactory) error {
+
+	// Check if object store is desired
+	desired, err := getObjectStoreState(context, false)
+	if err != nil {
+		return fmt.Errorf("failed to get desired state. %+v", err)
+	}
+
+	// Check if object store is applied
+	applied, err := getObjectStoreState(context, true)
+	if err != nil {
+		return fmt.Errorf("failed to get applied state. %+v", err)
+	}
+
+	if desired && !applied {
+		err := r.enable(context, factory)
+		if err != nil {
+			return fmt.Errorf("failed to enable the object store. %+v", err)
+		}
+
+	} else if !desired && applied {
+		err := r.remove(context, factory)
+		if err != nil {
+			return fmt.Errorf("failed to remove the object store. %+v", err)
+		}
+	}
+
+	return nil
+}
+
+// Configure the single instance of object storage in the cluster.
+func EnableObjectStore(context *clusterd.Context) error {
+	logger.Infof("Enabling object store")
+	key := path.Join(mon.CephKey, ObjectStoreKey, clusterd.DesiredKey)
+	_, err := context.EtcdClient.Set(ctx.Background(), key, "1", nil)
+	return err
+}
+
+// Remove the single instance of the object store from the cluster. All buckets will be purged..
```

Comment:
right now we're not purging any data when deleting the object store. Let's handle that with a separate issue. Seems like we would want a flag like `purge=true` to make sure they really want to get rid of the data.